backend/src/Designer/Helpers/PolicyConverter.cs

```diff
@@ -0,0 +1,241 @@
+                        List<string> resource = null;
+
+                        foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.Equals(XacmlConstants.MatchAttributeCategory.Subject)))
+                        {
+                            if (subject == null)
+                            {
+                                subject = new List<string>();
+                            }
+
+                            subject.Add($"{match.AttributeDesignator.AttributeId.ToString()}:{match.AttributeValue.Value}");
+                        }
+
+                        foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.Equals(XacmlConstants.MatchAttributeCategory.Resource)))
+                        {
+                            if (resource == null)
+                            {
+                                resource = new List<string>();
+                            }
+
+                            resource.Add($"{match.AttributeDesignator.AttributeId.ToString()}:{match.AttributeValue.Value}");
+
+                        }
+
+                        foreach (XacmlMatch match in allOf.Matches.Where(m => m.AttributeDesignator.Category.Equals(XacmlConstants.MatchAttributeCategory.Action)))
+                        {
+                            if (action == null)
+                            {
+                                action = new List<string>();
+                            }
+
+                            action.Add(match.AttributeValue.Value);
+                        }
+
+                        if (subject != null)
+                        {
+                            rule.Subject.AddRange(subject);
+                        }
+
+                        if (action != null)
+                        {
+                            rule.Actions.AddRange(action);
+                        }
+
+                        if (resource != null)
+                        {
+                            rule.Resources.Add(resource);
+                        }
+                    }
+
+                }
+
+                policy.Rules.Add(rule);
+            }
+
+            foreach(XacmlObligationExpression obligationExpression in xacmlPolicy.ObligationExpressions)
+            {
+                foreach (XacmlAttributeAssignmentExpression attributeAssignmentExpression in obligationExpression.AttributeAssignmentExpressions)
+                {
+                    if (attributeAssignmentExpression.Category.Equals(AltinnXacmlConstants.MatchAttributeCategory.MinimumAuthenticationLevel))
+                    {
+                        XacmlAttributeValue astr = attributeAssignmentExpression.Property as XacmlAttributeValue;
+                        policy.RequiredAuthenticationLevelEndUser = astr.Value;
+                    }
+
+                }
+            }
+
+            return policy;
+        }
+
+
+        public static XacmlPolicy ConvertPolicy(ResourcePolicy policyInput)
```

Comment:
At some point we need to validate that the `ResourcePolicy` supplied will result in valid XACML3, ie. it doesn't contain any empty lists for subjects, actions or resources (which will cause `<AnyOf />` in the serialized XML, which breaks spec as well as assumptions we make in the PDP).

Reply:
Added some first validaiton of that. Need to finalize this when looking in to what more we can validate. Added test